src/parser/rule/ruleset.rs

```diff
@@ -4,7 +4,7 @@ use super::*;
 pub struct Ruleset {}
 
 impl Parsable for Ruleset {
-    fn parse<I: Iterator<Item = char>>(parser: &mut Parser<I>) -> Result<Self, ParsingError> {
+    fn parse<I: Iterator<Item = char>>(_: &mut Parser<I>) -> Result<Self, ParsingError> {
```

Comment:
do we want to do this here or keep it until we implement this?

Reply:
I don't mind either way personally. I was just going to leave this alone tbh

Reply:
Personally if I would leave it `parser` but if the clippy errors get to annoying then we should do `_parser` not `_` IMO